src/components/Filters/FiltersDialog.tsx

```diff
@@ -0,0 +1,312 @@
+			: null;
+		onClose?.(filters);
+	};
+
+	const internalSchemaAndUiSchema = React.useMemo(() => {
+		const properties = schema.properties;
+		if (!properties) {
+			return undefined;
+		}
+
+		if (!Array.isArray(formData)) {
+			return {
+				schema: {
+					type: 'object',
+					properties: {
+						field: {
+							title: 'Field',
+							type: 'string',
+						},
+						operator: {
+							title: 'Operator',
+							type: 'string',
+						},
+						value: {
+							title: 'Value',
+							type: 'string',
+						},
+					},
+				},
+				uiSchema: {
+					'ui:grid': {
+						field: { xs: 4, sm: 4 },
+						operator: { xs: 4, sm: 4 },
+						value: { xs: 4, sm: 4 },
+					},
+					field: {
+						'ui:readonly': true,
+					},
+					operator: {
+						'ui:readonly': true,
+					},
+				},
+			};
+		}
+
+		const oneOf = Object.entries(properties)
+			/* since properties is of type JSONSchemaDefinition = JSONSchema | boolean,
+			 * we need to remove all possible boolean values
+			 */
+			.filter(
+				([_k, v]: [string, JSONSchemaDefinition]) => typeof v !== 'boolean',
+			)
+			.map(([key, property]: [string, JSONSchema]) => ({
+				title: property.title,
+				const: key,
+			}));
+
+		const uiSchema = {
+			'ui:ArrayFieldTemplate': ArrayFieldTemplate,
+			items: {
+				'ui:grid': {
+					field: { xs: 4, sm: 4 },
+					operator: { xs: 4, sm: 4 },
+					value: { xs: 4, sm: 4 },
+				},
+				value: {},
+			},
+		};
+
+		return {
+			schema: {
+				type: 'array',
+				minItems: 1,
+				items: formData?.map((data) => {
+					const schemaField: JSONSchema = {
+						$id: 'field',
+						title: 'Field',
+						type: 'string',
+						oneOf,
+					};
+					const { propertySchema, model } = getPropertySchemaAndModel(
+						data?.field ?? oneOf[0].const,
+						schema,
+					);
+					if (!model) {
+						return {};
+					}
+					const rendererSchema =
+						model.rendererSchema(schemaField, propertySchema, data) ?? {};
+					// This if statement is needed to display objects in a nice way.
+					// Would be nice to find a better way and keep schema and uiSchema separated
+					if (propertySchema?.type === 'object') {
+						if (!propertySchema.properties) {
+							return rendererSchema;
+						}
+						uiSchema.items.value = Object.fromEntries(
+							Object.entries(propertySchema.properties).map(
+								([key, value]: [string, JSONSchema]) => [
+									key,
+									{
+										'ui:title': '',
+										'ui:placeholder': value.title + '*',
+									},
+								],
+							),
+						);
+					}
+					return rendererSchema;
+				}),
+				additionalItems: {
+					field: formData?.[0].field,
+					operator: formData?.[0].operator,
+					value: undefined,
+				},
+			},
+			uiSchema,
+		};
+	}, [formData, schema]);
+
+	return !internalSchemaAndUiSchema ? null : (
+		<DialogWithCloseButton
+			fullWidth
+			title={t('labels.filter_by')}
+			onClose={() => {
+				setFormData(initialFormData);
+				onClose?.();
+			}}
+			open={open || !!formData}
```

Comment:
Remember that using `open` to display or hide does not actually render/un-render. If `open` is false, the Dialog is still rendered just not visible, so if there are any conflicts in the RJSForm or properties being passed into it, we may start hearing about it in the console. If you're sure there won't be any case where the form may crash/error while the dialog is not visible, it's okay to keep using open (though it will also put more load on the page since the page will immediately pre-render the dialog even though it's invisible). If you're not sure and it's possible some value may make it crash/break when the dialog is not supposed to be visible, then it would be better to conditionally display the Dialog: `{(open || !formData) && <DialogWithCloseButton ... />}`

Reply:
done in the previous change, set open to true, and add a condition on the component